hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/cache/CacheGetConfigMessageTask.java

```diff
@@ -19,25 +19,61 @@
 import com.hazelcast.cache.impl.operation.CacheGetConfigOperation;
 import com.hazelcast.client.impl.protocol.ClientMessage;
 import com.hazelcast.client.impl.protocol.codec.CacheGetConfigCodec;
+import com.hazelcast.client.impl.protocol.task.AbstractCallableMessageTask;
+import com.hazelcast.config.CacheConfig;
+import com.hazelcast.config.LegacyCacheConfig;
+import com.hazelcast.instance.BuildInfo;
 import com.hazelcast.instance.Node;
 import com.hazelcast.nio.Connection;
 import com.hazelcast.nio.serialization.Data;
-import com.hazelcast.spi.Operation;
+import com.hazelcast.spi.InternalCompletableFuture;
+import com.hazelcast.spi.impl.operationservice.InternalOperationService;
+import com.hazelcast.spi.properties.GroupProperty;
+
+import java.security.Permission;
+
+import static com.hazelcast.cache.impl.ICacheService.SERVICE_NAME;
 
 /**
- * This client request  specifically calls {@link CacheGetConfigOperation} on the server side.
+ * This client request specifically calls {@link CacheGetConfigOperation} on the server side.
  *
  * @see CacheGetConfigOperation
  */
 public class CacheGetConfigMessageTask
-        extends AbstractCacheMessageTask<CacheGetConfigCodec.RequestParameters> {
+        extends AbstractCallableMessageTask<CacheGetConfigCodec.RequestParameters> {
     public CacheGetConfigMessageTask(ClientMessage clientMessage, Node node, Connection connection) {
         super(clientMessage, node, connection);
     }
 
     @Override
-    protected Operation prepareOperation() {
-        return new CacheGetConfigOperation(parameters.name, parameters.simpleName);
+    protected Object call()
+            throws Exception {
+        final InternalOperationService operationService = nodeEngine.getOperationService();
+        CacheGetConfigOperation op = new CacheGetConfigOperation(parameters.name, parameters.simpleName);
+        InternalCompletableFuture future = operationService.invokeOnTarget(SERVICE_NAME,
+                op,
+                nodeEngine.getThisAddress());
+        return future.join();
```

Comment:
Can't we make this message task async too? Similar to `CacheCreateConfigMessageTask `...

Reply:
Not sure if it's applicable here but there is a `com.hazelcast.client.impl.protocol.task.AbstractAddressMessageTask` base class for invoking on target. For instance: https://github.com/hazelcast/hazelcast/blob/e0ea8ed98b3c30d4e8a03d38fc6fd9c6bca57574/hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/crdt/pncounter/PNCounterAddMessageTask.java

Reply:
Looks like a nice fit, thanks for the suggestion.